Rakefile.rb

```diff
@@ -48,6 +48,36 @@ def name_from(json)
   orgs.first[:name]
 end
 
+def file_to_commit_metadata
+  # There's about 2MB of data returned by this command, so rather than
+  # using backticks to turning it into a string and then splitting it,
+  # parse the output line-by-line. As a bonus this doesn't
+  # unnecessarily invoke a shell.
+  command = ['git', '--no-pager', 'log', '--name-only',  '--format=%h|%at', '--', 'data']
+  result = {}
+  sha, timestamp = nil
+  Kernel.open('|-', 'r') { |f|
+    if f
+      # In the parent process after forking, parse the output:
+      f.each_line { |line|
+        commit_match = line.match /^([a-f\d]+)\|(\d+)$/
+        if commit_match
+          sha, timestamp = commit_match.captures
+        else
+          filename = line.strip
+          unless result.has_key? filename
+            result[filename] = {sha: sha, timestamp: timestamp}
+          end
+        end
+      }
+    else
+      # In the child process after working, exec the command:
+      exec *command
+    end
+  }
+  result
```

Comment:
I think I'd be tempted to rename `result` to something that more adequately reflects _what_ it's actually storing, rather than just the fact that it's a holder for the return value. 

Maybe something like `last_commit`, so that the assignment reads:
```ruby
   last_commit[filename] ||= { … }
```
Something like that is likely to be a bit easier to grasp on an initial scan of the code before you've loaded the entire context of what this doing.